instrumentation/kafka-clients/kafka-clients-0.11/library/build.gradle.kts

```
@@ -0,0 +1,23 @@
+plugins {
+  id("otel.library-instrumentation")
+}
+
+val versions: Map<String, String> by project
+
+dependencies {
+  implementation(project(":instrumentation:kafka-clients:kafka-clients-common:library"))
+  library("org.apache.kafka:kafka-clients:2.8.1")
+
+  testImplementation(project(":instrumentation:kafka-clients:kafka-clients-0.11:testing"))
```

it looks like this works back to `2.6.0`, so let's make that the base version.

and let's update the module name to reflect, so `instrumentation/kafka-clients/kafka-clients-2.6/library`. it's ok for the javaagent and library instrumentation base versions not to match, we have some precedent for this already under `instrumentation/apache-httpclient`.

```suggestion
  implementation(project(":instrumentation:kafka-clients:kafka-clients-common:library"))
  library("org.apache.kafka:kafka-clients:2.6.0")

  testImplementation(project(":instrumentation:kafka-clients:kafka-clients-0.11:testing"))
  testImplementation("com.fasterxml.jackson.core:jackson-databind:2.10.2")
```

---

Why the extra `jackson-databind` ?

---

> is there any way for the user to configure the KafkaTracing that it uses?

There is `configure(Map)` method that gets called, if that would somehow help get the right `OpenTelemetry` instance ... ?

---

This one?
```java
void configure(Map<String, ?> configs);
```
As far as I understand, it gets passed (mostly) the same config the `KafkaProducer`/`KafkaConsumer` receives in its constructor - passing `OpenTelemetry` or `KafkaTracing` would still require a code change, it wouldn't work with config passed by env var/system property. And if you can change the code, you might as well use the `Consumer`/`Producer` wrappers exposed by `KafkaTracing`.
Overall I think it's best to recommend using the wrappers, and document clearly how the interceptor works and what are its limitations.

---

> Why the extra `jackson-databind` ?

Just skip it if it turns out to be not needed

---

> Overall I think it's best to recommend using the wrappers, and document clearly how the interceptor works and what are its limitations.

Yeah.
But I think folks using the interceptors already know all that -- it's nicely documented in the interfaces.
And when you use instantiation via string/class you shouldn't be too surprised by the limitations ...